MetaMask/metamask-extension
# EIP-1559 and Rinkeby Testnet support in Swaps

## Explanation
This PR:
- Adds support for EIP-1559 in Swaps
- Uses new components for editing gas in Swaps
- Adds support for a new network in Swaps (Rinkeby), because it already has EIP-1559 support

## Manual testing steps
EIP-1559 happy path:
- Switch to the `Rinkeby` Test Network
- Get test ETH
- Click on "Swap" on the main page
- Make a swap
- Open the transaction in a block explorer (e.g. https://rinkeby.etherscan.io/) and check that Txn Type is: `2 (EIP-1559)` 

Non EIP-1559 happy path:
- Use e.g. the Ethereum Mainnet Network
- Click on "Swap" on the main page
- Make a swap
- Open the transaction in a block explorer (e.g. https://etherscan.io/) and check that Txn Type is not: `2 (EIP-1559)` 

## Screenshots
`View Quote` page on Rinkeby, which has EIP-1559 support:
![image](https://user-images.githubusercontent.com/80175477/127176881-0d06d838-89b8-45d0-b3df-86a7054c126a.png)

Success page for an EIP-1559 transaction:
![image](https://user-images.githubusercontent.com/80175477/127176995-8c11626c-8f37-4d90-bee4-5854b9055e62.png)

`Customize Gas` modal window for EIP-1559, Basic Options:
![image](https://user-images.githubusercontent.com/80175477/127177727-ba42014f-725d-4326-9e54-07f357c2a885.png)

`Customize Gas` modal window for EIP-1559, Advanced Options:
![image](https://user-images.githubusercontent.com/80175477/127177797-ddf0624c-428b-410a-9c07-382b8503e16e.png)

EIP-1559 Txn Type on rinkeby.etherscan.io:
![image](https://user-images.githubusercontent.com/80175477/127177130-8b7e772c-9903-46ba-a52e-1b0adb497475.png)

EIP-1559 transaction in the `Activity` tab on the main page:
![image](https://user-images.githubusercontent.com/80175477/127177270-800d5723-896b-4068-a84c-7f21a589e718.png)

`View Quote` on Ethereum testnet, without EIP-1559 support:
![image](https://user-images.githubusercontent.com/80175477/127177462-6abaeb95-da00-4445-9eb7-4226b2363653.png)

Success page for a non-EIP 1559 transaction:
![image](https://user-images.githubusercontent.com/80175477/127177561-6d65440d-2baa-44d6-b05a-417590b34385.png)

Non-EIP-1559 transaction in the `Activity` tab on the main page:
![image](https://user-images.githubusercontent.com/80175477/127177596-8b406ea5-f79a-42c7-9908-ff226e04ff91.png)
